test/parsing.test.ts

```diff
@@ -839,10 +833,21 @@ describe('Zod Parsing', () => {
         a: z.string().default('hello'),
         b: z.number().default(() => 42),
         c: z.boolean().default(true),
-        d: z.null().default(),
+        d: z.null().default(null),
       });
       assert.deepEqual(schema.parse({}), { a: 'hello', b: 42, c: true, d: null });
-      assert.ok((schema as any)[coercionTypeSymbol]);
+    });
+
+    it('should return nested defaults with default object', () => {
```

Comment:
I added this test which is the use case we wanted. However I can't get the typing to work